### PR TITLE
Added Option to disable Witch Transform

### PR DESCRIPTION
**Pull request by TheShermanTanker**

### CODE REVIEW

**Comment by kennytv** [comment]
Spacing is still not quite right 👀

**Comment by TheShermanTanker** [comment]
> 
> 
> Spacing is still not quite right 👀

Fixed

Gimme a break it's confusing to edit through patch files :P

File: Spigot-Server-Patches/0552-Add-Option-to-disable-Witch-Transforms.patch [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+index e471e764935e2a89560de56959a782b02e5e8fe1..2a57a8a66a22a76853bb02e349a497c8765d77ec 100644
+--- a/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
++++ b/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
+@@ -72,6 +72,11 @@ public class PaperWorldConfig {
+         return config.getString("world-settings." + worldName + "." + path, config.getString("world-settings.default." + path));
+     }
+ 
++    public boolean witchConversion;
++    private void advancedSurvivalSettings() { //More may be added later
++    	witchConversion = getBoolean("advanced-survival-settings.witch-conversions", true);
```
Comment (kennytv) [inline Spigot-Server-Patches/0552-Add-Option-to-disable-Witch-Transforms.patch]: should be 4 spaces, not 1

File: Spigot-Server-Patches/0552-Add-Option-to-disable-Witch-Transforms.patch [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+     private void blockGrowthHeight() {
+diff --git a/src/main/java/net/minecraft/server/EntityVillager.java b/src/main/java/net/minecraft/server/EntityVillager.java
+index bf019043a9338aca8d91da809f1d5520531386e7..1590b430d6f86fd215b0108e2fccb24fe549ef3e 100644
+--- a/src/main/java/net/minecraft/server/EntityVillager.java
++++ b/src/main/java/net/minecraft/server/EntityVillager.java
+@@ -722,6 +722,7 @@ public class EntityVillager extends EntityVillagerAbstract implements Reputation
+ 
+     @Override
+     public void onLightningStrike(EntityLightning entitylightning) {
++    	if(!this.world.paperConfig.witchConversion) { super.onLightningStrike(entitylightning); return; } //Paper - Add Option to disable turning to Witch
```
Comment (kennytv) [inline Spigot-Server-Patches/0552-Add-Option-to-disable-Witch-Transforms.patch]: same as above

**Review by kennytv — CHANGES_REQUESTED** [review]
Nope :p

File: Spigot-Server-Patches/0552-Add-Option-to-disable-Witch-Transforms.patch (hunk 1 above)
Comment (TheShermanTanker) [reply Spigot-Server-Patches/0552-Add-Option-to-disable-Witch-Transforms.patch]: Wait, you meant that? I DID put 4 spaces though, why does it only show up as 1?

**Comment by TheShermanTanker** [comment]
Ok, edited the file one last time. Hopefully it's good now

**Comment by aikar** [comment]
I agree with the previous comments, that this belongs as a plugin and likely already exists.

While features are totally cool for paper, we typically reserve them to ones that a plugin can't do cleanly/without hacking internals....

This doesn't qualify for that, as it's literally 1 single event to cancel and takes 5 minutes to make.

**Comment by TheShermanTanker** [comment]
> 
> 
> I agree with the previous comments, that this belongs as a plugin and likely already exists.
> 
> While features are totally cool for paper, we typically reserve them to ones that a plugin can't do cleanly/without hacking internals....
> 
> This doesn't qualify for that, as it's literally 1 single event to cancel and takes 5 minutes to make.

I was thinking more about the people who aren't developers and aren't interested in searching for a plugin to disable 1 specific event

Regardless, I respect your decision

**Comment by aikar** [comment]
That's a dangerous slope to go down, as then we would end up putting every plugin into paper. 
People who arent developers commonly have to ask someone to help make/find simple plugins, part of running a server.

**Comment by BillyGalbreath** [comment]
[TestPlugin.zip](https://github.com/PaperMC/Paper/files/5059100/TestPlugin.zip) (rename this to .jar)

```
public class TestPlugin extends JavaPlugin {
    @Override
    public void onEnable() {
        getServer().getPluginManager().registerEvents(new Listener() {
            @EventHandler
            public void onLetsMakeAWitch(EntityTransformEvent event) {
                if (event.getTransformedEntity().getType() == EntityType.WITCH) {
                    event.setCancelled(true);
                }
            }
        }, this);
    }
}
```

**Comment by josephrooks** [comment]
And if you only need it in certain areas, WorldGuard has a region flag for it.

**Comment by wizjany** [comment]
news to me.

**Comment by pop4959** [comment]
pretty sure essentials protect has an option for this as well

**Comment by FazziCLAY** [comment]
Test

**Pull request closed without merging**